docs/self-hosting/guides/monitoring-telemetry.mdx

```diff
@@ -37,287 +39,282 @@ OTEL_TELEMETRY_COLLECTION_ENABLED=true
-       metrics_path: "/metrics"
-   ```
-
-   **Note**: Replace `infisical-backend:9464` with the actual hostname and port where your Infisical backend is running. This could be:
-
-   - **Docker Compose**: `infisical-backend:9464` (service name)
-   - **Kubernetes**: `infisical-backend.default.svc.cluster.local:9464` (service name)
-   - **Bare Metal**: `192.168.1.100:9464` (actual IP address)
-   - **Cloud**: `your-infisical.example.com:9464` (domain name)
-
-### Deployment Options
-
-#### Docker Compose
-
-```yaml
-services:
-  prometheus:
-    image: prom/prometheus:latest
-    ports:
-      - "9090:9090"
-    volumes:
-      - ./prometheus.yml:/etc/prometheus/prometheus.yml:ro
-    command:
-      - "--config.file=/etc/prometheus/prometheus.yml"
-
-  grafana:
-    image: grafana/grafana:latest
-    ports:
-      - "3000:3000"
-    environment:
-      - GF_SECURITY_ADMIN_USER=admin
-      - GF_SECURITY_ADMIN_PASSWORD=admin
 ```
 
-#### Kubernetes
-
-```yaml
-# prometheus-deployment.yaml
-apiVersion: apps/v1
-kind: Deployment
-metadata:
-  name: prometheus
-spec:
-  replicas: 1
-  selector:
-    matchLabels:
-      app: prometheus
-  template:
-    metadata:
-      labels:
-        app: prometheus
-    spec:
-      containers:
-        - name: prometheus
-          image: prom/prometheus:latest
+<Tabs>
+  <Tab title="Pull-based Monitoring (Prometheus)">
+    This approach exposes metrics on port 9464 at the `/metrics` endpoint, allowing Prometheus to scrape the data. The metrics are exposed in Prometheus format but originate from OpenTelemetry instrumentation.
+
+    ### Configuration
+
+    1. **Enable Prometheus export in Infisical**:
+
+       ```bash
+       OTEL_TELEMETRY_COLLECTION_ENABLED=true
+       OTEL_EXPORT_TYPE=prometheus
+       ```
+
+    2. **Expose the metrics port** in your Infisical backend:
+
+       - **Docker**: Expose port 9464
+       - **Kubernetes**: Create a service exposing port 9464
+       - **Other**: Ensure port 9464 is accessible to your monitoring stack
+
+    3. **Create Prometheus configuration** (`prometheus.yml`):
+
+       ```yaml
+       global:
+         scrape_interval: 30s
+         evaluation_interval: 30s
+
+       scrape_configs:
+         - job_name: "infisical"
+           scrape_interval: 30s
+           static_configs:
+             - targets: ["infisical-backend:9464"] # Adjust hostname/port based on your deployment
+           metrics_path: "/metrics"
+       ```
+
+       **Note**: Replace `infisical-backend:9464` with the actual hostname and port where your Infisical backend is running. This could be:
+
+       - **Docker Compose**: `infisical-backend:9464` (service name)
+       - **Kubernetes**: `infisical-backend.default.svc.cluster.local:9464` (service name)
+       - **Bare Metal**: `192.168.1.100:9464` (actual IP address)
+       - **Cloud**: `your-infisical.example.com:9464` (domain name)
+
+    ### Deployment Options
```

Comment:
feels a bit raw to jump right into the deployment options. what are these deployment options for? i think we need to give them more context prior to this

Reply:
updated